examples/contrib/readhar.py

```diff
@@ -0,0 +1,122 @@
+"""Reads HAR files into flow objects"""
+import asyncio
+import json
+import logging
+import time
+
+from mitmproxy import command
+from mitmproxy import connection
+from mitmproxy import ctx
+from mitmproxy import exceptions
+from mitmproxy import http
+from mitmproxy import types
+
+logger = logging.getLogger(__name__)
+
+
+class ReadHar:
+    def __init__(self):
+        pass
+
+    def fix_headers(
+        self, request_headers: list[dict[str, str]] | list[tuple[str, str]]
+    ) -> list[tuple[bytes, bytes]]:
+        """Converts provided headers into (b"header-name", b"header-value") tuples"""
+        flow_headers = []
+        for header in request_headers:
+            # Applications that use the {"name":item,"value":item} notation are Brave, Chrome, Edge, Firefox, Charles, Fiddler, Insomnia, Safari
+            if isinstance(header, dict):
+                key = header["name"]
+                value = header["value"]
+
+            # Application that use the [name, value] notation is Slack
+            else:
+                try:
+                    key = header[0]
+                    value = header[1]
+                except IndexError as e:
+                    raise exceptions.OptionsError(str(e)) from e
+            flow_headers.append((bytes(key, "utf-8"), bytes(value, "utf-8")))
+
+        return flow_headers
+
+    # Don't know how to make a type annotation for the request json
+    def request_to_flow(self, request_json: dict) -> http.HTTPFlow:
+        """
+        Creates a HTTPFlow object from a given entry in HAR file
+        """
+
+        request_method = request_json["request"]["method"]
+        request_url = request_json["request"]["url"]
+
+        request_headers = self.fix_headers(request_json["request"]["headers"])
+
+        client_conn = connection.Client(
+            peername=("127.0.0.1", 51513),
+            sockname=("127.0.0.1", 8080),
+            # TODO Get time info from HAR File
+            timestamp_start=time.time(),
+        )
+
+        # 375:3 is default mitmproxy server_conn when making a new flow.
+        server_conn = connection.Server(address=("375", 3))
+
+        new_flow = http.HTTPFlow(client_conn, server_conn)
+        try:
+            new_flow.request = http.Request.make(
+                request_method, request_url, "", request_headers
+            )
+        except TypeError as e:
+            logger.error("Failed to create request")
+            raise exceptions.OptionsError(str(e)) from e
```

Comment:
As with https://github.com/mitmproxy/mitmproxy/pull/6190#discussion_r1243518517, I think we don't need the `try` block here, it does not provide any additional context when it would happen. I'm also not even sure why/how `http.Request.make` would even fail here.